src/client/mod.rs

```diff
@@ -101,6 +103,8 @@ struct ClientInner {
     topology: Topology,
     options: ClientOptions,
     session_pool: ServerSessionPool,
+    #[cfg(feature = "fle")]
+    fle: std::sync::Mutex<Option<fle::ClientState>>,
```

Comment:
It's a little awkward to have to wrap this in a `Mutex` since it's not mutated after client construction, but since making the state for FLE needs the main `Client` as part of its own initialization parameters, I didn't see a way around this.

Reply:
We could potentially get around this by creating a `WeakClient` type or some such, but I think we're going to need some form of lock anyways in order to invoke the mutable mongocrypt methods, so it's not a big deal. Since `ClientState` stores `Client`s within it, we need to be really careful not to use them while holding the `std` lock guard, as we can easily deadlock (maybe the compiler will enforce this?). To avoid this, we could introduce a wrapper type or something that manages handling the lock, or alternatively just use an async one.

Also, we may want a `RwLock`, where the read portion can be used to access the clients and the write portion for mutating the state. Though maybe its best to defer a decision on these questions until the implementation moves a little further and the use cases are clearer.

Reply:
I realized this needed to be a weak ref in any case, since otherwise there's a potential for creating a reference loop with the main `Client` holding a ref to itself.  I also updated it to a `RwLock` because even very early further implementation points to that being the way to go.